### PR TITLE
Add followers in the frontend.

### PR DESCRIPTION
**Pull request by JonathanOsAlc**

### CODE REVIEW

**Comment by JonathanOsAlc** [comment]
@moylop260, now in the frontend you can follow or unfollow the builds.

When the user follows the build:
image

When the user unfollows the build:
image

File: runbot_send_email/static/src/js/runbot.js [hunk 1]
```diff
@@ -1,44 +1,24 @@
-odoo.define('runbot.test', function (require) {
+odoo.define('orchestv.actions', function (require) {
```
Comment (moylop260) [inline runbot_send_email/static/src/js/runbot.js]: orchestv?

Comment (oscarolar) [reply runbot_send_email/static/src/js/runbot.js]: haha, copypasteerror

Comment (JonathanOsAlc) [reply runbot_send_email/static/src/js/runbot.js]: @moylop260 haha sorry, modified.

**Comment by moylop260** [comment]
@JonathanOsAlc 
Do you know what means `runbot_send_email/static/src/js/runbot.js:16: [W7903(javascript-lint), ] Parsing error: Unexpected token > [Error]`?

https://travis-ci.org/Vauxoo/runbot-addons/jobs/359460430#L735

**Comment by moylop260** [comment]
- [ ] Could you remove the method that creates partner from commiter_email of git?
- [ ] Could you auto add a follower if the commiter_email of git is matching with a partner.email that exists?
- [ ] Remove all methods to get data from html email format and migrate it to html format directly
E.g. delete the method `_shareissue_link` and use `https://github.com/Vauxoo/runbot-addons/issues/new` in the html template email directly

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 2]
```diff
@@ -121,3 +121,16 @@ def test_60_coverage_value_error_form(self):
     def test_70_coverage_value_error_template(self):
         self.env.ref('runbot_send_email.runbot_send_notif').unlink()
         self.build._github_status()
+
+    def test_80_update_followers(self):
+        """Test for the method update_followers.
+        """
+        user = self.env['res.users'].browse(self.env.uid)
+        result = self.build.update_followers()
+        followers = self.build.message_follower_ids.mapped('partner_id').ids
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: It's not required use `.ids`
You can use browser directly is falsy too

File: runbot_send_email/tests/test_runbot_send_email.py [hunk 3]
```diff
@@ -121,3 +121,16 @@ def test_60_coverage_value_error_form(self):
...
+        self.assertFalse(result)
+        self.assertFalse(followers)
+        result = self.build.update_followers()
+        followers = self.build.message_follower_ids.mapped('partner_id').ids
```
Comment (moylop260) [inline runbot_send_email/tests/test_runbot_send_email.py]: It's not required use `.ids`

File: runbot_send_email/models/runbot_build.py [hunk 4]
```diff
@@ -165,82 +165,20 @@ def _shareissue_link(self):
...
-                record.with_context(email_ctx).message_post_with_template(
-                    email_ctx.get('default_template_id'))
-                _logger.info('Sent email to: %s, Build: %s', email_to,
-                             name_build)
-
     def _github_status(self):
         super(RunbotBuild, self)._github_status()
         for record in self:
             record.get_github_links()
-            record.send_email()
```
Comment (moylop260) [inline runbot_send_email/models/runbot_build.py]: Return the `super` result, please 
It's not your code but we could fix it here

**Comment by oscarolar** [comment]
>@JonathanOsAlc
>Do you know what means runbot_send_email/static/src/js/runbot.js:16: [W7903(javascript-lint), ] Parsing error: Unexpected token > [Error]?
>
>https://travis-ci.org/Vauxoo/runbot-addons/jobs/359460430#L735

@moylop260 this happens because @JonathanOsAlc is using ES6 (ECMAScript 2015) notation, his code translates into:
**Traditional JS**
```javascript
var sumMethod = function(a, b){
    return a + b;
}
```

**ES6**
```javascript
var sumMethod = (a, b) => a + b;
```
They are called _arrow functions_.

There are two ways to fix this lint, whether @JonathanOsAlc make its _Traditional JS_ or the linters are updated to support **ES6**

**Comment by JonathanOsAlc** [comment]
@moylop260, now in the frontend, we have a button that allows to the user `watch` or `unwatch` the repository, and if the user watches the repository, automatically the user will be following all builds that belong the repository.
image
image

And the email only will be sent when the build has in state `running`:
image

File: runbot_send_email/data/runbot_send_email_data.xml [hunk 5]
```diff
@@ -222,8 +181,8 @@ Build #${object.sequence} is ${object.state} but failed
           <td style="padding:0px;border:0px">
           </td>
           <td style="padding:0px;border:0px;color:#808080">
-              <a href="${object.doc_link}" style="text-decoration:underline;color:#606060" target="_blank">Documentation</a> about <span class="il">Vauxoo's Runbot</span> and <a href="${object.dockerdoc_link}" style="text-decoration:underline;color:#606060" target="_blank">generated docker</a>. <br>
-                  For help please send a message to Vauxoo's telegram group, choose who receive this notification in your <a href="${object.configfile_link}" style="text-decoration:underline;color:#606060" target="_blank">Config File</a>.
+              <a href="/runbot_doc/static/index.html" style="text-decoration:underline;color:#606060" target="_blank">Documentation</a> about <span class="il">Vauxoo's Runbot</span> and <a href="https://github.com/Vauxoo/travis2docker/wiki" style="text-decoration:underline;color:#606060" target="_blank">generated docker</a>. <br>
```
Comment (moylop260) [inline runbot_send_email/data/runbot_send_email_data.xml]: where is this ur?

File: runbot_send_email/data/runbot_send_email_data.xml [hunk 6]
```diff
@@ -172,7 +131,7 @@ Build #${object.sequence} is ${object.state} but failed
           </p>
         </span>
     </div>
-% elif (object.state=="running" or object.state=="done") and (object.result=="warn"):
+% elif object.result=="warn":
```
Comment (moylop260) [inline runbot_send_email/data/runbot_send_email_data.xml]: what about ko result?

File: runbot_send_email/models/runbot_build.py [hunk 7]
```diff
@@ -106,141 +97,60 @@ def _subject_email(self):
...
+            follower = False
+        return follower
+
+    def create(self, vals):
+        """Add the followers of repository to the followers of build.
+        """
+        build_id = super(RunbotBuild, self).create(vals)
+        users = build_id.repo_id.message_partner_ids.mapped('user_ids')
+        build_id.message_subscribe_users(user_ids=users.ids)
+        return build_id
```
Comment (moylop260) [inline runbot_send_email/models/runbot_build.py:151]: Where is the code that adds the git.commiter as follower is the there are a user.email?

File: runbot_send_email/data/runbot_send_email_data.xml (hunk 5 above)
Comment (JonathanOsAlc) [reply runbot_send_email/data/runbot_send_email_data.xml]: @moylop260 the URL is complete with the host before it send the email.
image

File: runbot_send_email/data/runbot_send_email_data.xml (hunk 6 above)
Comment (JonathanOsAlc) [reply runbot_send_email/data/runbot_send_email_data.xml]: @moylop260, it is the `else` in this [line](https://github.com/vauxoo-dev/runbot-addons/blob/bce30ed1bde6f846add1d6c14c4b5ae4a37bdd1e/runbot_send_email/data/runbot_send_email_data.xml#L90).

File: runbot_send_email/models/runbot_build.py (hunk 7 above)
Comment (JonathanOsAlc) [reply runbot_send_email/models/runbot_build.py:151]: @moylop260, I do it when sending the email in this [lines](https://github.com/vauxoo-dev/runbot-addons/blob/bce30ed1bde6f846add1d6c14c4b5ae4a37bdd1e/runbot_send_email/models/runbot_build.py#L105-L115), but I think that is better move it to the method `create`, right?

**Comment by moylop260** [comment]
Superseded by https://github.com/Vauxoo/runbot-addons/pull/161
(Because a hard sync of oca/11.0 branch was executed)

**Pull request closed without merging**